Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -750,7 +750,18 @@ static int previousSentenceStartFromPosition(StringView text, unsigned position)
 
 static int nextSentenceEndFromPosition(StringView text, unsigned position)
 {
-    return ubrk_following(sentenceBreakIterator(text), position);
+    int endPosition = ubrk_following(sentenceBreakIterator(text), position);
+
+    if (!text.substring(position, endPosition).containsOnly<isASCIIWhitespace>()) {
+        // To match AXObjectCache::nextBoundary, don't include a newline character at the end of sentences.
+        while (endPosition > 0 && text.length() && text.substring(0, endPosition).endsWith('\n'))
+            endPosition -= 1;
+    } else {
+        // If we are looking at a range that is *only* newline characters, the end should be the next sentence boundary.
+        while (endPosition < (int)text.length() - 1 && text.length() && text.substring(0, endPosition + 1).endsWith('\n'))
+            endPosition += 1;
+    }
+    return endPosition;
 }
```

Comment:
I would rather use offset or index instead of position for the naming here, since "position" is used in this context for instances of Position and VisiblePosition classes.